src/lib.rs

```diff
@@ -285,4 +463,33 @@ mod tests {
 
         Ok(())
     }
+
+    fn number_digit_addition_prop<'a, D>(
+        a: &'a [u32],
+        b: D,
+        b_u32: u32,
```

Comment:
Third argument can be eliminated:

```rust
    fn number_digit_addition_prop<'a, D>(a: &'a [u32], b: D) -> Result<(), TestCaseError>
    where
        D: Copy,
        for<'b> &'b BigInt: std::ops::Add<D, Output = BigInt>,
        for<'b> D: std::ops::Add<&'b BigInt, Output = BigInt>,
        for<'b> &'b num_bigint::BigUint: std::ops::Add<D, Output = num_bigint::BigUint>,
    {
        // We assume that `a` is non-negative
        prop_assume!(a.is_empty() || (a.last().unwrap() >> 31) == 0);

        let a_gmp = BigInt::from_digits_u32(a).reallocate();
        let result_gmp_1 = &a_gmp + b;
        let limbs_gmp_1 = result_gmp_1.to_digits_u32();
        let result_gmp_2 = b + &a_gmp;
        let limbs_gmp_2 = result_gmp_2.to_digits_u32();

        let a_num = num_bigint::BigUint::new(a.to_vec());
        let result_num = &a_num + b;
        let limbs_num = result_num.to_u32_digits();

        prop_assert_eq!(strip_padding_u32(&limbs_gmp_1), limbs_num.clone());
        prop_assert_eq!(strip_padding_u32(&limbs_gmp_2), limbs_num);

        Ok(())
    }
```

Reply:
We'll have to delete `add_number_and_digit` test, but it's not much different from `add_number_and_u32`

Reply:
If you're OK with deleting this test, then this is surely way to go